From a3931ec6f43857aeed7feac5d223d7db6728145e Mon Sep 17 00:00:00 2001
From: Povilas Kanapickas <povilas@radix.lt>
Date: Thu, 25 Mar 2021 22:46:33 +0200
Subject: [PATCH] xfree86: Create a symlink X -> Xorg when installing using
 meson

This brings the behavior closer than what we currently have with
autotools-based build system.

Meson does not currently have native support for symlinks.
See https://github.com/mesonbuild/meson/issues/1602.

Signed-off-by: Povilas Kanapickas <povilas@radix.lt>
---
 hw/xfree86/meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hw/xfree86/meson.build b/hw/xfree86/meson.build
index a6311cb12..b1b5920a5 100644
--- a/hw/xfree86/meson.build
+++ b/hw/xfree86/meson.build
@@ -138,6 +138,12 @@ if get_option('xf86-input-inputtest')
     subdir('drivers/inputtest')
 endif
 
+meson.add_install_script(
+    'sh', '-c',
+    'ln -fs Xorg @0@@1@'.format(
+        '${DESTDIR}',
+        join_paths(get_option('prefix'), get_option('bindir'), 'X')))
+
 if get_option('suid_wrapper')
     executable('Xorg.wrap',
         'xorg-wrapper.c',
-- 
GitLab

